openfl/component/director/director.py

```diff
@@ -172,38 +171,37 @@ def remove_experiment_data(self, experiment_name: str, caller: str):
         if experiment_name in self.experiment_stash.get(caller, {}):
             del self.experiment_stash[caller][experiment_name]
 
-    def collaborator_health_check(
-            self, *, collaborator_name: str, is_experiment_running: bool
+    def envoy_health_check(
+            self, *, envoy_name: str, is_experiment_running: bool,
+            cuda_devices_status: list = None,
     ) -> int:
         """Accept health check from envoy."""
-        shard_info = self._shard_registry.get(collaborator_name)
+        shard_info = self._shard_registry.get(envoy_name)
         if not shard_info:
-            raise Exception(f'Unknown shard {collaborator_name}')
+            raise Exception(f'Unknown shard {envoy_name}')
 
         hc_period = self.settings.get('envoy_health_check_period', ENVOY_HEALTH_CHECK_PERIOD)
         shard_info['is_online']: True
         shard_info['is_experiment_running'] = is_experiment_running
         shard_info['valid_duration'] = 2 * hc_period
         shard_info['last_updated'] = time.time()
 
+        if cuda_devices_status is not None:
+            for i in range(len(cuda_devices_status)):
+                shard_info['shard_info']['node_info']['cuda_devices'][i] = cuda_devices_status[i]
+
         return hc_period
 
     def get_envoys(self) -> list:
         """Get a status information about envoys."""
         logger.info(f'Shard registry: {self._shard_registry}')
-        envoy_infos = []
-        for envoy in self._shard_registry.values():
-            envoy_info = director_pb2.EnvoyInfo(
-                shard_info=envoy['shard_info'],
-                is_online=time.time() < envoy['last_updated'] + envoy['valid_duration'],
-                is_experiment_running=envoy['is_experiment_running']
+        for envoy_info in self._shard_registry.values():
+            envoy_info['is_online'] = (
+                time.time() < envoy_info['last_updated']
+                + envoy_info['valid_duration']
```

Comment:
Could `envoy_info['last_updated'] + envoy_info['valid_duration']` be on the one line in order to increase readability?